lib/xandra/connection.ex

```diff
@@ -757,25 +754,30 @@ defmodule Xandra.Connection do
     end
   end
 
-  defp handle_frame(%__MODULE__{} = data, %Frame{stream_id: stream_id} = frame) do
+  defp handle_frame(
+         %__MODULE__{timed_out_ids: timed_out_ids} = data,
+         %Frame{stream_id: stream_id} = frame
+       ) do
     case pop_in(data.in_flight_requests[stream_id]) do
-      {nil, data} ->
-        if Map.has_key?(data.timed_out_ids, stream_id) do
-          :telemetry.execute(
-            [:xandra, :timed_out_response],
-            telemetry_meta(data, %{stream_id: stream_id})
-          )
+      # There is no in-flight req for this response frame, BUT there is a request
+      # for it that timed out on the caller's side. Let's just emit a
+      {nil, data} when is_map_key(timed_out_ids, stream_id) ->
+        :telemetry.execute(
+          [:xandra, :debug, :received_timed_out_response],
```

Comment:
For now, let's keep this a "debug" event so that you folks can measure this but we don't have to make it part of the public API.